kernel/src/last_checkpoint_hint.rs

```diff
@@ -37,6 +37,11 @@ pub(crate) struct LastCheckpointHint {
 }
```

Comment:
can't comment far outside the diff but seems like we might be able to make the const private (remove `pub(crate)`) if only usage is now just in that new function?

Reply:
I think this is addressed now.